### DISCUSSION TITLE
Simple E-Ink display

### DISCUSSION THREAD

**Discussion by zacgoodwin** [outdoorbits/little-backup-box#332] [Ideas]
Howdy! 

After doing a lot of search for the perfect hardware for this project I've come down on using a Pi zero W 2 with waveshare 2.13 e-ink touch screen and a USB2Go SD card reader and 1 TB microSD card. The benefit of this is that you can power it with a regular portable battery and you don't need an external hard drive for the storage. When you get back you can just pop it into your computer and go. 

That being said; I suck with software. I was wondering if anyone had any knowledge to do something simple to display a couple things on the eInk screen
- Ready for backup
- Backing up
- Backup Completed

This is the documentation: https://www.waveshare.com/wiki/2.13inch_Touch_e-Paper_HAT_Manual#Install_Python_Libraries

I'm willing to help how I can with hardware or whatever is needed. 
Thanks

**Comment by outdoorbits** [comment]
This display looks fantastic! Actually I'm a bit too busy to test it myself. For an integration in lbb it would be best and cleanest if it was supported by luma https://github.com/rm-hull/luma.core. In this case at least most of the lbb display handing could be untouched and all the functionality would be preserved. Could you find out what driver chip is used in your display and if this is supported by luma? Maybe it would be worth to ask them to support this display as they really are experts in that.

**Comment by zacgoodwin** [comment]
I am not sure if it is supported by Luma.lcd; The waveshare seems to be based on a ST7735 which Luma says it supports. LBB seems to see the display natively (though I can't set the resolution correctly and not sure if the touchscreen works correctly); it is 

![image](https://github.com/user-attachments/assets/59ac99b9-e36d-441b-ba3a-a5d73bb59882)

I check into Luma, I didn't see any documentation about e-Ink displays. They started down that path with https://github.com/rm-hull/luma.lcd/issues/61 but it seems to have stalled. The waveshare seems to be a regular SPI device and in that thread it was mentioned that adding SPI devices is pretty easy for them so like you said, doesn't hurt to ask.

It looks like the waveshare screen does have it's own python libraries: Touch_e-Paper_Code\python\lib\TP_lib\ 
Which can be found here: https://github.com/waveshare/Touch_e-Paper_HAT

After that it looks like that library enables https://pillow.readthedocs.io/en/stable/ for ImageDraw to be used the same that Luma does. 

I am happy to foot any development costs / hardware costs. What I'd like to end up with is a finished configuration that I can image so I can make these quickly for my friends.  Print a little container, wire up a battery underneath with an on off button, and you have something like the beedrive except actually awesome.

**Comment by outdoorbits** [comment]
LBB does not detect screens or their resolution, the values are just 
possible values for the known displays. So in a first step, I would add 
your driver and if we are lucky, that's it. I hope I will find some time 
this weekend.

**Comment by outdoorbits** [comment]
As the driver was already supported, I just added the resolution options 
for 250x122 pixels. Because the e-ink display has a low refresh rate, 
you will have to set DISPLAY -  "Minimum display time per frame in 
seconds" to 2 seconds or more. Does it work?

Unfortunately, the display menu will not support the touch screen.

**Comment by zacgoodwin** [comment]
Sorry for the delay; I am just getting back to this project. I'm having a bit of trouble but I think I need to spend some more time with the wiki.

**Reply by outdoorbits** [reply]
I'm still excited. What are the problems?

**Reply by zacgoodwin** [reply]
I wasn't able to get it to display anything and it didn't seen to recognize when an SD card was inserted. Didn't seem to recognize any of the readers I used either. I wonder if it's because it's an OTG port on the 0?

Going to play around to find out. If you know of any logs that would be good to post here and have a second set of eyes let me know :D

**Comment by zacgoodwin** [comment]
Howdy! 

I was never able to get this working; would you like me to mail it to you to see if you have better luck? 

I got the OLED one from the tutorials but running into heat issues I'm working on solving.

**Reply by outdoorbits** [reply]
Maybe this isn't necessary. Because an actual issue involving nearly the same driver some charges  could make it working. 
Could you please update and have a look at the wiki article about this driver?

**Reply by outdoorbits** [reply]
Would you please try the display using the last version of lbb?

**Comment by outdoorbits** [comment]
Are we talking about this one: [https://www.berrybase.de/2.13-250-122-epaper-display-hat-fuer-raspberry-pi](https://www.berrybase.de/2.13-250-122-epaper-display-hat-fuer-raspberry-pi)?

**Comment by outdoorbits** [comment]
ST7735 is running now. Could you please test your display? It it is still not working, you could send me one if you still like. I would try to support it by lbb.
